examples/node/cli.rs

```diff
@@ -941,57 +958,90 @@ fn list_offers(node: &LightningNode) -> Result<()> {
 
     println!("Total of {} offers\n", offers.len().to_string().bold());
     for offer in offers {
-        let kind = match offer.offer_kind {
-            OfferKind::Pocket { .. } => "Pocket",
-        };
+        print_offer(&offer);
+        println!();
+    }
 
-        let created_at: DateTime<Local> = offer.created_at.into();
-        let expires_at: Option<DateTime<Local>> = offer.expires_at.map(|e| e.into());
-
-        println!("{kind} offer created at {created_at}:");
-        println!("      Expires at:         {expires_at:?}");
-        println!(
-            "      Amount:             {}",
-            amount_to_string(offer.amount)
-        );
-        println!("      LNURL-w:            {:?}", offer.lnurlw);
-        match offer.offer_kind {
-            OfferKind::Pocket {
-                id,
-                exchange_rate,
-                topup_value_minor_units,
-                exchange_fee_minor_units,
-                exchange_fee_rate_permyriad,
-                error,
-                ..
-            } => {
-                println!("                   ID:    {id}");
-                println!(
-                    "      Value exchanged:    {:.2} {}",
-                    topup_value_minor_units as f64 / 100f64,
-                    exchange_rate.currency_code,
-                );
-                println!(
-                    "      Exchange fee rate:  {}%",
-                    exchange_fee_rate_permyriad as f64 / 100_f64
-                );
-                println!(
-                    "      Exchange fee value: {:.2} {}",
-                    exchange_fee_minor_units as f64 / 100f64,
-                    exchange_rate.currency_code,
-                );
-                let exchanged_at: DateTime<Utc> = exchange_rate.updated_at.into();
+    Ok(())
+}
+
+fn print_offer(offer: &OfferInfo) {
+    let kind = match offer.offer_kind {
+        OfferKind::Pocket { .. } => "Pocket",
+    };
```

Comment:
Should we implement `Display` for `Offerkind` instead? Or is it that you don't want to bloat the Core library with implementations that are only being used in the CLI tool..?

Reply:
Yes, I would say that how we display things in the CLI is very specific to the CLI itself.